plugins/scaffolder/src/next/TemplateWizardPage/Stepper/Stepper.tsx

```diff
@@ -61,12 +68,38 @@ export const Stepper = (props: StepperProps) => {
     );
   }, [props.extensions]);
 
+  const validators = useMemo(() => {
+    return Object.fromEntries(
+      props.extensions.map(({ name, validation }) => [name, validation]),
+    );
+  }, [props.extensions]);
+
+  const validator = useMemo(() => {
+    const { mergedSchema } = steps[activeStep];
+    return createAsyncValidator(mergedSchema, validators, {
+      apiHolder,
+    });
+  }, [steps, activeStep, validators, apiHolder]);
+
   const handleBack = () => {
     setActiveStep(prevActiveStep => prevActiveStep - 1);
   };
 
-  const handleNext = ({ formData }: { formData: JsonObject }) => {
-    setActiveStep(prevActiveStep => prevActiveStep + 1);
+  const handleNext = async ({ formData }: { formData: JsonObject }) => {
+    setErrors(undefined);
```

Comment:
probably not already here, right? should it just mark the form as "busy" (greyed out and whatnot) and then update the errors as needed at the end?

Reply:
Yeah loading states are a little unsettled yet, as maybe each component could have it's own loading state. Leaving that for now, but yes 💯